src/data/finset/basic.lean

```diff
@@ -1504,6 +1504,231 @@ theorem forall_mem_insert [d : decidable_eq α]
   (∀ x, x ∈ insert a s → p x) ↔ p a ∧ (∀ x, x ∈ s → p x) :=
 by simp only [mem_insert, or_imp_distrib, forall_and_distrib, forall_eq]
 
+lemma eq_or_ne {α : Type*} [D : decidable_eq α] {x y : α} : (x = y)∨ (x ≠ y) :=
+begin
+  have A1:decidable (x=y) := D x y,
+  cases A1,
+  {
+    right,
+    apply A1,
+  },
+  {
+    left,
+    apply A1,
+  }
+end
+
+lemma lt_or_not_lt {α:Type*} [D:linear_order α] {x y:α}:(x<y)∨ ¬ (x< y) :=
+begin
+  have A1:decidable (x<y) := linear_order.decidable_lt x y,
+  cases A1,
+  {
+    right,
+    apply A1,
+  },
+  {
+    left,
+    apply A1,
+  }
+end
+
+lemma le_or_not_le {α : Type*} [D : linear_order α] {x y : α} : (x ≤ y) ∨ ¬(x ≤ y) :=
+begin
+  have A1:decidable (x≤y) := linear_order.decidable_le x y,
+  cases A1,
+  {
+    right,
+    apply A1,
+  },
+  {
+    left,
+    apply A1,
+  }
+end
+
+
+lemma le_iff_not_gt {α:Type*} [linear_order α] {a b:α}:a ≤ b ↔ ¬ a > b :=
+begin
+  split,
+  {
+    intro A1,
+    intro A2,
+    apply not_le_of_gt A2,
+    apply A1,
+  },
+  {
+    apply le_of_not_gt,
+  }
+end
+
+lemma has_le_fun_def {α β:Type*} [preorder β] {f g:α → β}: f ≤ g = (∀ a:α, f a ≤ g a) := rfl
+
+
+lemma le_func_def2 {α β:Type*} [preorder β] {f g:α → β}:(f ≤ g) ↔ (∀ n:α, f n ≤ g n) :=
```

Comment:
We should keep this version if it doesn't already exist (the proof is `iff.rfl`)

Reply:
It's `pi.le_def`